src/testtool/sep/septool_unix_win/sepperuse.1
.TH SEPPERUSE 1 "rti" "Relational Technology, Inc." "Relational Technology, Inc."
.\" History
.\"	17-jan-1990 (rog)
.\"		Written.
.\"	24-jan-1990 (boba)
.\"		Add -a, -d and -o options and multiple .cfg files.  
.\"		Add scanning for DIFF and how to get to top.
.\"		Move menu options from OPTIONS to DESCRIPTION.
.\"	02-dec-1991 (boba)
.\"		Change "delete" menu option to "ok".  Fix references
.\"		to diff files (should be log files).
.\"	27-dec-1991 (boba)
.\"		General audit tool man page cleanup.
.ta 8n 16n 24n 32n 40n 48n 56n
.SH NAME
sepperuse \- Examine SEP test log files for tests with differences.
.SH ORIGIN
Relational Technology, Inc.
.SH SYNOPSIS
.B sepperuse
[
.B -a ] [
.B -d ] [
.B -o
.I outdir
] [ 
.I LISTEXEC-config-file\fB.cfg\fP ...
]
.SH DESCRIPTION
.I Sepperuse
displays each of the SEP test log files from the specified LISTEXEC
configuration file(s) that contain differences.
.PP
If no \fB.cfg\fP file is specified, \fIsepperuse\fP
will check all of the \fB*.out\fP files
in the \fB$ING_TST/output\fP directory for tests with differences and display
the \fB*.log\fP files from those tests.
.PP
If the
.B -a
option is specified, then all \fB*.log\fP files will be displayed,
regardless of the \fB.cfg\fP or \fB*.out\fP files.
.PP
The
.B -o
option may be used to override the default output directory,
\fB$ING_TST/output\fP.  Alternatively, if the environment variable
\fB$tst_output\fP is set, this will be used instead of \fB$ING_TST/output\fP.
.PP
Each SEP log file is displayed via \fImore\fP, or via \fIpg\fP if 
\fImore\fP is not available.
After a file has been displayed, the user is given a menu of the following
options:
.TP
.B h)elp
Print this message.
.TP
.B n)ext
Go to next log file, leaving this for later perusal.
.TP
.B r)epeat
Show this log file again.
.TP
.B p)rint
Print the log file with
.B lpr -p.
.TP
.B o)k
Mark log file as OK in \fB$ING_TST/output/SEPOK\fP so it won't be seen again.
.TP
.B s)hell
Launch a \fB$SHELL\fP, \fIcsh\fP default.
.TP
.B b)ye
Exit \fIsepperuse\fP.
.PP
If the
.B -d
option is specified,
the \fImore\fP or \fIpg\fP command is set up to scan for
the first occurance of "DIFF" (display by \fIsep\fP of a difference).
With \fImore\fP, you can go back to the top of the file by entering
"'" or continue to the next "DIFF" by entering "n".
With \fIpg\fP, you can go back to the top of the file by entering
"?Test Name?^CR" or continue to the next "DIFF" by entering "/^CR".
.SH OPTIONS
.TP
.B -a
display all \fB*.log\fP files whether they are mentioned
in a corresponding \fB*.out\fP file or not.
.TP
.B -d
scan for first occurence of "DIFF" in \fB*.log\fP file.
.TP
.B -o \fIoutdir\fP
search for \fB*.out\fP and \fB*.log\fP files in \fIoutdir\fP rather than the
default \fB$ING_TST/output\fP.
.SH FILES
.SH "SEE ALSO"
.BR sep(1),
.BR listexec(1),
.BR updperuse(1)
.SH BUGS
.I Sepperuse
requires that \fB$ING_TST\fP and \fB$TST_LISTEXEC\fP be set.


src/testtool/sep/septool_unix_win/sepperuse.sh
:
#
# Copyright (c) 2004 Ingres Corporation
#
# sepperuse -- peruse sep tests with differences
#
# History:
#	17-jan-1990 (rog)
#		Written.
#	24-jan-1990 (boba)
#		Allow for override of default output directory with -o
#		option or by setting $tst_output.  Peruse all .log
#		files regardless of the .cfg or .out files with the -a
#		option.  Add usage message.  Allow for multiple .cfg
#		files.  Make more & pg search for "DIFF" if -d is 
#		specified.  Misc. fixes.
#	25-jan-1990 (boba)
#		Fix bug in creation of list for -a option (don't include
#		full path).  The Delete option wasn't deleting files.
#		Also, remove "-f" for delete option so it TELLS you if
#		something is wrong.
#	02-dec-1991 (boba)
#		Remove "delete" option and, instead, add "ok" option to
#		keep track of valid differences during audit.  Ok list
#		is kept in $ING_TST/output/SEPOK.  Also, clean up help
#		message still based on qaperuse diffs, not sep log files.
##      01-Aug-2004 (hanch04)
##          First line of a shell script needs to be a ":" and
##          then the copyright should be next.

: must be set ${ING_TST?} ${TST_LISTEXEC?}

if more < /dev/null > /dev/null 2>&1
then
	more=more
	diffscan="+/DIFF"
else
	if pg < /dev/null > /dev/null 2>&1
	then
		more=pg
		diffscan="+/DIFF/"
	else
		echo "Cannot find MORE or PG - using CAT."
		more=cat
	fi
fi

# check the args
alllog=false
moreparam=""
while [ $# != 0 ]
do
    case $1 in
	-a) alllog=true
	    ;;
	-d) moreparam="$diffscan"
	    ;;
	-o) shift
	    ( cd $1 ) || exit 1
	    tst_output=`cd $1; pwd`
	    ;;
	-*) echov "usage: sepperuse [ -a ] [ -o <outdir> ] [ <cfg-file-name> ... ]"
	    exit 255
	    ;;
	*)  files="$files $*"; break
	    ;;
    esac
    shift
done

[ "$files" = "" ] && nocfg=true || nocfg=false


# set directory & file symbols
tst_output=${tst_output-$ING_TST/output}
sepok=$tst_output/SEPOK
touch $sepok

cd $tst_output

if $alllog
then
    list="`ls *.log`"
else
    allout=/tmp/peruse.$$
    if $nocfg
    then
	cat $tst_output/*.out > $allout
    else
        for file in $files
        do
	    [ -r $TST_LISTEXEC/$file ] || {
	        echov "$TST_LISTEXEC/$file does not exist!"
	        exit 1
	    }

	    out=`echov $file | sed 's/.cfg/.out/'`

	    if [ -r $out ]
	    then
		cat $out >> $allout
	    else
	        echov "Cannot find $out SEP output file.  Have you run the $file tests?"
	        exit 1
	    fi
        done
    fi

    list=`awk '
        /\=\>/ {
	    while (index($4,"/") != 0) {
	        $4 = substr($4,index($4,"/")+1)
	    }
	    $4 = substr($4,1,index($4,"."))
	    save = sprintf("%slog",$4)
        }

        /Number of diff/ {
	    if ($6 != 0)
	        print save
        }
    ' < $allout`

    rm -f $allout
fi

for i in $list
do
    if grep "$i" $sepok > /dev/null
    then
	redo=false
    else
	redo=true
    fi
    while $redo
    do
	redo=false
	echov "\n--- begin ($tst_output/$i) ---"
	$more $moreparam $i
	echov "--- end ($tst_output/$i) ---"
	prompting=true
	while $prompting
	do
	    prompting=false
	    echov "\nh)elp n)ext r)epeat p)rint o)k s)hell b)ye: \c"
	    read desire
	    case $desire in
		[hH]*)
		    cat << !

h)elp		Print this message.
n)ext		Go to next log file, leaving this for later perusal.
r)epeat		Show this log file again.
p)rint		Print the log file with lpr -p.
o)k		Mark log file as OK in $sepok so it won't be seen again.
s)hell		Launch a $SHELL, csh default.
b)ye		Exit sepperuse.
!
		    prompting=true
		    ;;
		[bB]*)
		    exit
		    ;;
		[sS]*)
		    ${SHELL-/bin/csh}
		    echov "sepperuse: continuing..."
		    redo=true
		    ;;
		[oO]*)
		    echo "$i" >> $sepok
		    ;;
		[pP]*)
		    (lpr -p $tst_output/$i &)
		    redo=true
		    ;;
		[rR]*)
		    redo=true
		    ;;
		*)
		    ;;
	    esac
	done # prompting
    done # redo
done # file
